### PR TITLE
Fix/style/adapt light and dark mode

### PR DESCRIPTION
**Pull request by francelu**
# Adapt app to light and dark mode

## Description
This PR introduces our app's responsiveness to dark and light mode. It closes issue #182, subtask from #178.  
It also solves a few issues found out while working.

## Changes
- Try to reduce uncovered to by moving from `app/src/main/java/com/android/periodpals/ui/theme/` to `app/src/main/java/com/android/periodpals/resources/` :
  - `Color.kt`,
  - `Dimens.kt`,
  - and `Type.kt`.
- Clean up unused variables about `userState` in `SignInScreen` and `SignUpScreen`.
- Adapted colours in all UI screens.

#### Moved
From : `app/src/main/java/com/android/periodpals/ui/theme/`  
To : `app/src/main/java/com/android/periodpals/resources/`  
Reason: trying to augment code coverage (as this resources package might not be checked by SonarCloud).

#### Modified
- `Color.kt`:
  - New `object` `ComponentColor` which contains some "default" colour settings to be called in components for `color = ...` (usually).
  - Deleted old hard-coded pink and purple values.
- `Type.kt`: fixed compact M screen's profile picture size to `200.dp`, was `50.dp`
- `Theme.kt`:
  - Removed toast messages for screen size and related variables.
- `SignInScreen.kt`: Removed unused `userState` parameter and related variables from `attemptSignIn()`.
- `SignUpScreen.kt`: Removed unused `userState` parameter and related variables from `attemptSignUp()`.
- All UI screens have adapted colours parameters.

## Dependencies Added
- None.

## Testing
- Verified on emulators to ensure consistent and responsive design in both light and dark mode.

## Screenshots

| Screen                               | Before Light                                                                                                      | After Light                                                                                                       | Before Dark                                                                                                      | After Dark                                                                                                       |
|--------------------------------------|-------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------|
| **`SignInScreen`**                   | image                         | image                         | image                         | image                         |
| **`SignUpScreen`**                   | image                         | image                         | image                         | image                         |
| **`CreateProfileScreen`**            | image                         | image                         | image                         | image                         |
| **`ProfileScreen`**                  | image                         | image                         | image                         | image                         |
| **`EditProfileScreen`**              | image                         | image                         | image                         | image                         |
| **`CreateAlertScreen`**              | image                         | image                         | image                         | image                         |
| **`AlertListsScreen` - My alerts (empty)** | image                         | image                         | image                         | image                         |
| **`AlertListsScreen` - My alerts (non-empty)** | image                         | image                         | image                         | image                         |
| **`AlertListsScreen` - Pals' alerts (empty)** | image                         | image                         | image                         | image                         |
| **`AlertListsScreen` - Pals' alerts (non-empty)** | image                         | image                         | image                         | image                         |
| **`MapScreen`**                      | image                         | image                         | image                         | image                         |
| **`TimerScreen`**                    | image                         | image                         | image                         | image                         |

### CODE REVIEW

**Review by coaguila — APPROVED** [review]
lgtm

File: app/src/main/java/com/android/periodpals/resources/Color.kt [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+          surfaceContainerLowest = Color(0xFF0C0E13),
+          surfaceContainerLow = Color(0xFF191C20),
+          surfaceContainer = Color(0xFF1D2024),
+          surfaceContainerHigh = Color(0xFF282A2F),
+          surfaceContainerHighest = Color(0xFF33353A))
+}
+
+object ComponentColor {
+  /** Returns the primary card colors for alert items. */
+  @Composable
```
Comment (lazarinibruno) [inline app/src/main/java/com/android/periodpals/resources/Color.kt:100]: **Question**: why are these functions needed? Isn't the theme automatically applied? 

I understand that in some places you do:

```kotlin
colors = getPrimaryCardColors()
```

But I thought that having the theme defined would automatically do this.

File: app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt [hunk 2]
```diff
@@ -84,7 +82,6 @@ fun SignInScreen(
     navigationActions: NavigationActions,
 ) {
   val context = LocalContext.current
-  val userState: UserAuthenticationState by authenticationViewModel.userAuthenticationState
```
Comment (lazarinibruno) [inline app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt:87]: **Question**: why did you remove this?

**Review by lazarinibruno — APPROVED** [review]
## Summary

I really like the new colors 🌈🌈 Especially the gradients in the loading screen are just _chef's kiss_.

## Improvements

Now that you are at it, is there a way to hide the system bar where the notification appear? Or maybe adapt the color to match the top app bar?

If you have time it would be a nice addition to this PR. However, I know you have midterm tomorrow. So don't worry about it if you don't have time, we can do this in another moment.

## Steps before merge
Please look at my comments. Otherwise, all gud.

File: app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt (hunk 2 above)
Comment (francelu) [reply app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt:87]: As I had discussed with @charliemangano, we missed that on the PR reviews from the fix double click.  

As this parameter was no longer used since we moved the authentication logic from the UI back into VM part, I deleted it as well as I was touching this screen. This is to keep general code consistency and cleanness since we do not want to have to do clean up tasks in the future.

**Comment by francelu** [comment]
> I really like the new colors 🌈🌈 Especially the gradients in the loading screen are just _chef's kiss_.

Luvely heheh :))

**Comment by francelu** [comment]
> ## Improvements
> 
> Now that you are at it, is there a way to hide the system bar where the notification appear? Or maybe adapt the color to match the top app bar?
> 
> If you have time it would be a nice addition to this PR. However, I know you have midterm tomorrow. So don't worry about it if you don't have time, we can do this in another moment.

@lazarinibruno  
Good point, however I consider out of this scope (since I have my midterm tomorrow haha). However you can add it as a "For Later" task in the Scrum board :)  
Thank you for your consideration about my exam :))

File: app/src/main/java/com/android/periodpals/resources/Color.kt (hunk 1 above)
Comment (francelu) [reply app/src/main/java/com/android/periodpals/resources/Color.kt:100]: Uhm nup, it don't think so.
Example: component `Button` can have many purposes which require either primary, secondary or tertiary colours depending on it's display, importance and functionality.  

Maybe I'm wrong, but that's how I understood things worked. For example, we use `Card` for an alert item AND for a `NoAlertDialog`. However, their importance isn't the same; the first takes `primaryContainer` colours and the second one takes `tertiaryContainer`.  

I don't know if that's clear enough, it's hard to explain. Maybe there's an easier way. I don't have much time to figure that out.  

Main point is that now dark theme is readable, and there is no hard coded colours.

Comment (lazarinibruno) [reply app/src/main/java/com/android/periodpals/resources/Color.kt:100]: Yes, what you say totally makes sense. I had not thought of that situation. Thanks for the clarification.

File: app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt (hunk 2 above)
Comment (lazarinibruno) [reply app/src/main/java/com/android/periodpals/ui/authentication/SignIn.kt:87]: Okay, gotcha!

**Comment by lazarinibruno** [comment]
Perfect! Its all set then. Just fix the conflicts and you are good to go.
